Server/ConcordiaCurriculumManager/Services/CourseService.cs

```diff
@@ -370,28 +370,32 @@ public async Task<ICollection<CourseVersion>> GetCourseVersions(Dossier dossier)
 
     private async Task VerifyCourseIsNotInCourseGroupingOrThrow(Dossier dossier, Course course)
     {
-        var grouping = await _courseGroupingRepository.GetCourseGroupingContainingCourse(course);
+        var groupings = await _courseGroupingRepository.GetCourseGroupingsContainingCourse(course);
+        groupings = groupings?.Where(grouping => grouping.Published && !grouping.State.Equals(CourseGroupingStateEnum.Deleted)).ToList();
 
-        if (grouping is null)
+        if (groupings is null || groupings.Count == 0)
         {
             return;
         }
 

```

Comment:
`GetCourseGroupingsContainingCourse` returns groupings with state Accepted and with state Deleted. I think at this line you should filter out the deleted ones. I haven't tested it, but I am imagine if you create a deletion request for grouping x that contains course y, then accept that dossier so the grouping is deleted, then in a different dossier attempt to delete course y, the system will prevent it.

Reply:
The deleted ones are already filtered out in the previous line:
`groupings = groupings?.Where(grouping => grouping.Published && !grouping.State.Equals(CourseGroupingStateEnum.Deleted)).ToList();`

Unless you mean I should not filter out the groups that are not published.

Reply:
Oh yes sorry I missed that, but it did bring to mind that you're filtering for only published groupings. In general, we don't consider the publish state at all in these requests, so I think that should be removed (for example, imagine one user creates a grouping but it's not yet published (it doesn't get published until the end of the year usually), but then another user wants to delete courses that were in that grouping. With the current system, he's able to delete that course even though it would put the system into an inconsistent state).